Documentation/Enums/Enum Declarations.md

```diff
@@ -0,0 +1,223 @@
+# Enum Declarations
+
+An enumeration, or enum for short, is a static listing of named values.
+Enums are a quick and efficient method of storing a limited range of possibilities as integers.
+
+
+## Commands
+
+### `enum end`
+
+`enum end`
+
+Ends declaring an enum type.
+
+### `enum start`
+
+`enum start` `:` `name`
+
+Starts declaring an enum type.
+
+### `enum member`
+
+`enum member` `:` `name value`*`[, ","]`*
+
+Declares a value of an enum inside its declaration. The value must be a numeric.
+
+A comma may end the line to indicate this is not the last enum member in its declaration. 
+
+
+## Usage
+
+```
+enum start : Direction
+    enum member : Unknown 0 ,
+    enum member : Horizontal 1 ,
+    enum member : Vertical 2
+enum end
+```
+
+### CSharp
+
+```
+enum Direction {
+    Unknown = 0,
+    Horizontal = 1,
+    Vertical = 2
+}
+```
+
+### Java
+
+```
+enum Direction {
+    Unknown (0),
+    Horizontal (1),
+    Vertical (2)
+}
```

Comment:
Unfortunately, Java's a little more complicated. Enums can have constructors and take any number of values, but you have to implement the functionality yourself. An example:

```Java
enum Number {
   One(1), Two(2), Three(3);
  
   int numberValue;
   Number(int v) {
      numberValue = v;
   }
   int getValue() {
      return numberValue;
   } 
}

// Then where you use it, you need to decide whether you want the enum type or the value, and can't convert implicitly.
int value = Number.One.getValue();
Number one = Number.One;
int valueFail = Number.One; // no good, need Number.One.getValue()
```